### ISSUE TITLE
Improvements for file encryption/decryption

### ISSUE THREAD

**Issue by dschuermann** [open-keychain/open-keychain#665]
### Improvements
- UI: Make it a TextField with an attachment icon on the left when something is selected, use new open icon from Android (as now used in import activity)
- Show sdcard in side menu in storage API? I have seen this elsewhere, this should be possible somehow...
- Batch-encryption? with multi-select in storage API?
- Propose filename with gpg ending when storing it using the storage API?
- Don't show output dialog on Android 4.4, it is only used to provide a way for inputting a output filename, with the storage API this should not be necessary, just directly encrypt/decrypt
### Backward compatibility for Android < 4.4
- OK should also work with images selected from the gallery, not just with files selected from a file manager. Does this now work? (We should never try to parse the Uri, just use it for getting the input stream)
- Also only show filename, not the whole path after selecting a file on Android < 4.4 (to have the same activity design on all Android versions).
### Being a Document Provider

A more extended approach would be to let OK be a [Document Provider](https://developer.android.com/guide/topics/providers/document-provider.html). This would be displayed like Google Drive for example, but being a local storage for encrypted files.
Basically, Android's idea is to go away from a file based storage model. Instead it aims to be a app based storage model.

There is actually already a finished example where a storage provider is doing encryption decryption. See https://plus.google.com/+JeffSharkey/posts/9BmGb3xbPcA

It would be extremly awesome if I could just go into any application that uses the storage API and open an image from the OK storage which is then decrypted on the fly (maybe we could even ask the user if he wants to decrypt it or use the encrypted file directly).

**As always, these points are open for discussion**
